### PR TITLE
Fix DataGrid sorting for initialInput 'DESC' 

### PR DESCRIPTION
**Pull request by rdonigian**

### CODE REVIEW

**Comment by CarsonF** [comment]
I was bothered by the null chaining here. We know we have a sort item, so I updated types to reflect that.

I also thought that your logic was a bit hard to follow. And it is by chance that the sort selection order is asc -> desc -> null. I think this could be flipped around separately. So I think what I have is more clear on what we are actually doing and ensure we flip the order correctly regardless of grid configuration.

What do you think?

**Comment by rdonigian** [comment]
> I was bothered by the null chaining here. We know we have a sort item, so I updated types to reflect that.
> 
> I also thought that your logic was a bit hard to follow. And it is by chance that the sort selection order is asc -> desc -> null. I think this could be flipped around separately. So I think what I have is more clear on what we are actually doing and ensure we flip the order correctly regardless of grid configuration.
> 
> What do you think?

**Comment by rdonigian** [comment]
I like that approach, I was fighting the optional chaining.  LGTM

**Review by CarsonF — APPROVED** [review]
